docs/internals/variable_cleanup.rst

```diff
@@ -1,29 +1,31 @@
 .. index: variable cleanup
 
 *********************
-Cleaning Up Variables
+Zeroing of Padding
 *********************
 
-When a value is shorter than 256 bit, in some cases the remaining bits
-must be cleaned.
-The Solidity compiler is designed to clean such remaining bits before any operations
-that might be adversely affected by the potential garbage in the remaining bits.
-For example, before writing a value to  memory, the remaining bits need
-to be cleared because the memory contents can be used for computing
-hashes or sent as the data of a message call.  Similarly, before
-storing a value in the storage, the remaining bits need to be cleaned
-because otherwise the garbled value can be observed.
+When a value is shorter than 256 bit, in some cases the remaining (padding) bits
+must be zeroed. The Solidity compiler is designed to zero any padding bits before operations
+that might be adversely affected by garbage in the padding bits.
+For example, before writing a value to  memory, the padding bits need
+to be cleared because the memory contents could be used for computing
+hashes or sent as the data of a message call. Similarly, before
+storing a value in storage, padding bits need to be cleaned
+because otherwise their value can be observed, and may leak information.
 
-Note that access via inline assembly is not considered such an operation:
-If you use inline assembly to access Solidity variables
-shorter than 256 bits, the compiler does not guarantee that
-the value is properly cleaned up.
+Note that inline assembly does not automatically zero padding bits
+of values used therein.
```

Comment:
```suggestion
```
Isn't this redundant? The following lines mention this.

Reply:
This sentence is about what the assembly itself does. The following lines are about what the compiler does.

I'll clean up the language when I clean up the prior issue.